Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,9 +138,173 @@ public class PushDeploymentController : ApiController
             }
         }
 
-        private string GetZipURLFromARMJSON(JObject requestObject)
+        [HttpPost]
+        [HttpPut]
+        public async Task<HttpResponseMessage> OnePushDeploy(
+            [FromUri] string type = null,
+            [FromUri] bool async = false,
+            [FromUri] string path = null,
+            [FromUri] bool clean = false,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("OnePushDeploy"))
+            {
+                var response = Request.CreateResponse();
+
+                var website_framework = System.Environment.GetEnvironmentVariable(Constants.Framework).ToLower();
+
+                var deploymentInfo = new ArtifactDeploymentInfo(_environment, _traceFactory)
+                {
+                    AllowDeploymentWhileScmDisabled = true,
```

Comment:
Some of the properties being defined here are not changing the default value of those properties. In such cases, it would be good to skip them.